include/envoy/upstream/outlier_detection.h

```diff
@@ -140,17 +141,18 @@ class EventLogger {
    * @param type supplies the type of the event.
    * @param enforced is true if the ejection took place; false, if only logging took place.
    */
-  virtual void logEject(HostDescriptionConstSharedPtr host, Detector& detector, EjectionType type,
+  virtual void logEject(const HostDescriptionConstSharedPtr host, Detector& detector,
+                        envoy::data::cluster::v2alpha::OutlierEjectionType type,
                         bool enforced) PURE;
 
   /**
    * Log an unejection event.
    * @param host supplies the host that generated the event.
    */
-  virtual void logUneject(HostDescriptionConstSharedPtr host) PURE;
+  virtual void logUneject(const HostDescriptionConstSharedPtr host) PURE;
```

Comment:
actually this and other SharedPtr can be passed by const ref.

Reply:
ping on this, can you pass by const reference, here and elsewhere.